cot/src/router/path.rs

```diff
@@ -313,30 +371,60 @@ mod tests {
     #[test]
     #[should_panic(expected = "Consecutive parameters are not allowed")]
     fn path_parser_consecutive_params() {
-        let _ = PathMatcher::new("/users/:id:post_id");
+        let _ = PathMatcher::new("/users/{id}{post_id}");
     }
 
     #[test]
     #[should_panic(expected = "Invalid parameter name: ``")]
     fn path_parser_invalid_name_empty() {
-        let _ = PathMatcher::new("/users/:");
+        let _ = PathMatcher::new("/users/{}");
     }
 
     #[test]
     #[should_panic(expected = "Invalid parameter name: `123`")]
     fn path_parser_invalid_name_numeric() {
-        let _ = PathMatcher::new("/users/:123");
+        let _ = PathMatcher::new("/users/{123}");
     }
 
     #[test]
     #[should_panic(expected = "Invalid parameter name: `abc#$%`")]
     fn path_parser_invalid_name_non_alphanumeric() {
-        let _ = PathMatcher::new("/users/:abc#$%");
+        let _ = PathMatcher::new("/users/{abc#$%}");
+    }
+
+    #[test]
+    #[should_panic(expected = "Unclosed parameter: `foo`")]
+    fn path_parser_unclosed() {
+        let _ = PathMatcher::new("/users/{foo");
+    }
+
+    #[test]
+    #[should_panic(expected = "Closing brace encountered without opening brace")]
+    fn path_parser_missing_opening_brace() {
+        let _ = PathMatcher::new("/users/foo}");
+    }
+
+    #[test]
+    #[should_panic(expected = "Unclosed parameter: `foo`")]
+    fn path_parser_unclosed_slash() {
+        let _ = PathMatcher::new("/users/{foo/bar");
+    }
+
+    #[test]
+    #[should_panic(expected = "Unclosed parameter: `foo`")]
+    fn path_parser_unclosed_double() {
+        let _ = PathMatcher::new("/users/{foo{bar");
+    }
```

Comment:
Please add a test with `/users/{{{foo}}/bar` or similar to ensure that escaping and unclosed parameter check work at the same time

Reply:
Done